containerd-shim-v2/service.go

```diff
@@ -701,11 +701,13 @@ func (s *service) Update(ctx context.Context, r *taskAPI.UpdateTaskRequest) (*pt
 	defer s.Unlock()
 
 	var resources specs.LinuxResources
-	if err := json.Unmarshal(r.Resources.Value, &resources); err != nil {
-		return empty, err
+	v, err := typeurl.UnmarshalAny(r.Resources)
+	if err != nil {
+		return nil, err
 	}
+	resources = *v.(*specs.LinuxResources)
```

Comment:
Please add a switch case to check for the type, and if this is not the expected type, let's throw an error.

Reply:
updated.